lib/request-handler.js

```diff
@@ -79,13 +80,18 @@ function handleRequest (makeBadge, handlerOptions) {
   return (queryParams, match, end, ask) => {
     const reqTime = new Date();
 
+    let maxAge = envFlag(process.env.LONG_CACHE, false) ? 900 : 0;
     if (queryParams.maxAge !== undefined && /^[0-9]+$/.test(queryParams.maxAge)) {
-      ask.res.setHeader('Cache-Control', 'max-age=' + queryParams.maxAge);
-      ask.res.setHeader('Expires', new Date(+reqTime + queryParams.maxAge * 1000).toGMTString());
-    } else {
-      // Cache management - no cache, so it won't be cached by GitHub's CDN.
+      // always let queryParams.maxAge override the default
+      // regardless of env.LONG_CACHE setting
+      maxAge = queryParams.maxAge;
+    }
+    if (maxAge === 0) {
       ask.res.setHeader('Cache-Control', 'no-cache, no-store, must-revalidate');
-      ask.res.setHeader('Expires', reqTime.toGMTString());  // Proxies, GitHub, see #221.
+      ask.res.setHeader('Expires', reqTime.toGMTString());
+    } else {
+      ask.res.setHeader('Cache-Control', 'max-age=' + maxAge);
+      ask.res.setHeader('Expires', new Date(+reqTime + maxAge * 1000).toGMTString());
```

Comment:
It's explained here https://github.com/badges/shields/pull/1651#issuecomment-382939663

Reply:
Thanks for that pointer!

From https://github.com/badges/shields/pull/1651#issuecomment-382939663:

> It may be pointless in a perfect world but different implementations take different approach, there will always be someone not following the spec or maybe also not supporting Cache-Control but supporting Expires.

It seems like we're acting on a kind of superstition. It would be great to have an example of a client which does this, or a reliable report that this is true.

I'm fine leaving it in for now, though let's add a reference that stays with the code.

Reply:
The Cache-Control header was defined as part of the HTTP/1.1 specification would be ignored by any HTTP client which implements the HTTP/1.0 standard but not HTTP/1.1. This is definitely an edge case and we should expect any modern web browser or downstream cache to implement HTTP/1.1 but we can't safely assume that _only_ clients implementing HTTP/1.1 will connect to us. I've added a comment clarifying this.

Reply:
Thanks for explaining that!